orca-core/src/main/java/com/netflix/spinnaker/orca/locks/LockingConfigurationProperties.java

```diff
@@ -0,0 +1,53 @@
+package com.netflix.spinnaker.orca.locks;
+
+import com.netflix.spinnaker.kork.dynamicconfig.DynamicConfigService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.context.properties.ConfigurationProperties;
+import org.springframework.stereotype.Component;
+
+@Component
+@ConfigurationProperties("locking")
+public class LockingConfigurationProperties {
+  private boolean learningMode = true;
+  private boolean enabled = false;
+  private int ttlSeconds = 120;
+  private int backoffBufferSeconds = 10;
+  private final DynamicConfigService dynamicConfigService;
+
+  @Autowired
+  public LockingConfigurationProperties(DynamicConfigService dynamicConfigService) {
+    this.dynamicConfigService = dynamicConfigService;
+  }
+
+  public boolean isLearningMode() {
+    return dynamicConfigService.getConfig(Boolean.class, "locking.learningMode", learningMode);
```

Comment:
I don't entirely hate this pattern of injecting the `DynamicConfigService` into the `@ConfigurationProperties` object

Reply:
I like it.